pynicotine/shares.py

```diff
@@ -477,11 +477,11 @@ def is_hidden(folder, filename=None, entry=None):
         """Stop sharing any dot/hidden folders/files."""
 
         # If the last folder in the path starts with a dot, or is a Synology extended
-        # attribute folder, we exclude it
+        # attribute, snapshot or recycle bin folder, we exclude it
         if filename is None:
             last_folder = os.path.basename(folder)
 
-            if last_folder.startswith(".") or last_folder == "@eaDir":
+            if last_folder.startswith(".") or last_folder in ["@eaDir", "#recycle", "#snapshot"]:
```

Comment:
I would make the folder list a class constant, such as:
`HIDDEN_FOLDER_NAMES = {"@eaDir", "#recycle", "#snapshot"}`

then access it with

`if last_folder.startswith(".") or last_folder in self.HIDDEN_FOLDER_NAMES:`

Reply:
Done

Reply:
No worries!